lib/appsignal/utils/push_api_key_validator.ex

```diff
@@ -0,0 +1,12 @@
+defmodule Appsignal.Utils.PushApiKeyValidator do
+  def validate(config) do
+    HTTPoison.start
+    url = "#{config[:endpoint]}/1/auth?api_key=#{config[:push_api_key]}"
+    case HTTPoison.get url do
+      {:ok, %HTTPoison.Response{status_code: 200}} -> :ok
+      {:ok, %HTTPoison.Response{status_code: 401}} -> :invalid
```

Comment:
It is not really idiomatic to return `:invalid` here when you also return `:ok` and `{:error, ...}` tuples. When returning `:ok` (or an `:ok`-tuple) it is generally assumed that you either return this or an error tuple.
So maybe return `{:error, :invalid}` instead?